terraform/ec2/win/main.tf

```diff
@@ -34,66 +58,67 @@ resource "aws_instance" "cwagent" {
-</powershell>
-EOF
 
   tags = {
-    Name = "cwagent-integ-test-ec2-windows-${element(split("/", var.test_dir), 3)}-$${module.common.testing_id}"
+    Name = "cwagent-integ-test-ec2-windows-${module.common.testing_id}"
   }
 }
 
 resource "null_resource" "integration_test" {
   depends_on = [aws_instance.cwagent]
+
   # Install software
+  connection {
+    type     = "winrm"
+    user     = "Administrator"
+    password = rsadecrypt(aws_instance.cwagent.password_data, local.private_key_content)
+    host     = aws_instance.cwagent.public_dns
+  }
+
+  provisioner "file" {
+    source      = "${var.test_dir}/${local.final_validator_config}"
+    destination = "${local.instance_temp_directory}/${local.final_validator_config}"
+  }
+
+  provisioner "file" {
+    source      = "${var.test_dir}/${local.cloudwatch_agent_config}"
+    destination = "${local.instance_temp_directory}/${local.cloudwatch_agent_config}"
+  }
+
+  # Install agent binaries
   provisioner "remote-exec" {
     inline = [
-      "start /wait timeout 120",                             //Wait some time to ensure all binaries have been downloaded
-      "call %ProgramData%\\chocolatey\\bin\\RefreshEnv.cmd", //Reload the environment variables to pull the latest one instead of restarting cmd
-      "set AWS_REGION=${var.region}",
+      "git clone --branch ${var.github_test_repo_branch} ${var.github_test_repo}",
       "aws s3 cp s3://${var.s3_bucket}/integration-test/packaging/${var.cwa_github_sha}/amazon-cloudwatch-agent.msi .",
       "start /wait msiexec /i amazon-cloudwatch-agent.msi /norestart /qb-",
-      "echo clone and install agent",
-      "git clone ${var.github_test_repo}",
+
+    ]
+  }
+
+  provisioner "remote-exec" {
+    inline = [
       "cd amazon-cloudwatch-agent-test",
-      "git reset --hard ${var.cwa_test_github_sha}",
-      "echo run tests with the tag integration, one at a time, and verbose",
-      "echo run sanity test && go test ./test/sanity -p 1 -v",
-      "go test ${var.test_dir} -p 1 -timeout 30m -v"
+      "go run ./validator/main.go --validator-config=${local.instance_temp_directory}/${local.final_validator_config} --preparation-mode=true",
     ]
+  }
 
-    connection {
-      type            = "ssh"
-      user            = "Administrator"
-      password        = rsadecrypt(aws_instance.cwagent.password_data, local.private_key_content)
-      host            = aws_instance.cwagent.public_ip
-      target_platform = "windows"
-      timeout         = "6m"
-    }
+  provisioner "remote-exec" {
+    inline = [
+      "set AWS_REGION=${var.region}",
+      "cd amazon-cloudwatch-agent-test",
+      "powershell \"& 'C:\\Program Files\\Amazon\\AmazonCloudWatchAgent\\amazon-cloudwatch-agent-ctl.ps1' -a fetch-config -m ec2 -s -c file:${local.instance_temp_directory}/${local.cloudwatch_agent_config}\"",
+      "go run ./validator/main.go --validator-config=${local.instance_temp_directory}/${local.final_validator_config} --preparation-mode=false",
```

Comment:
Why use the validator over the dir directly, since this runs on host. I don't really see the value of a validator over running a test directly for ec2 testing.

Reply:
It would be the same comment as this one https://github.com/aws/amazon-cloudwatch-agent-test/pull/166#discussion_r1135680435